src/LinAlg/CMakeLists.txt

```diff
@@ -2,6 +2,7 @@
 # Set headers to be installed as part of the hiop interface
 set(hiopLinAlg_INTERFACE_HEADERS
   hiop_blasdefs.hpp
+  hiopLinAlgFactory.hpp
```

Comment:
This should not be in HiOp API. @CameronRutherford, are you sure you need this here? This header is needed to build HiOp library, but not HiOp applications.

Reply:
For as long as `hiopMatrixMDS` implementation is all in header files, `hiopLinAlgFactory.hpp` needs to be installed. This needs to be handled in a separate issue; out of scope of this PR.